_layouts/home.html

```diff
@@ -2,6 +2,23 @@
 layout: archive
 ---
 
+{% if page.url == "/" %}
+  {% assign entries_layout = page.entries_layout | default: 'list' %}
+  {% assign print_header = true %}
+  <div class="entries-{{ entries_layout }}">
+    {% assign posts = site.posts %}
+    {% for post in posts %}
+      {% if post.pinned %}
```

Comment:
This is redundant and slow. Use [where expression](https://jekyllrb.com/docs/liquid/filters/):

```liquid
{% assign posts = site.posts | where: 'pinned', true %}
```

Reply:
@iBug I'll look at this on my personal site. I'm not a pro at ruby/jekyll, but I am a dev. Thanks